### PR TITLE
Update docs for tiered storage

### PR DESCRIPTION
**Pull request by Anonymitaet**
Currently, the docs for tiered storage is not clear and incomplete.

This PR adds and reorganize contents for tiered storage.

Old structure:
image

New structure:
image

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@gaoran10 @Huanli-Meng could u pls help review? Thanks

File: site2/docs/tiered-storage-aws.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+  
+- Apache jclouds: 2.2.0 or later versions
+
+### Step
+
+This example uses Pulsar 2.5.1.
+
+1. Download Pulsar tarball using one of the following ways:
+
+   * download from the [Apache mirror](https://archive.apache.org/dist/pulsar/pulsar-2.5.1/apache-pulsar-2.5.1-bin.tar.gz)
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-aws.md:25]:
```suggestion
   * Download the Pulsar tarball from the [Apache mirror](https://archive.apache.org/dist/pulsar/pulsar-2.5.1/apache-pulsar-2.5.1-bin.tar.gz).
```
same comments for the next bullet

File: site2/docs/tiered-storage-aws.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+- Pulsar: 2.4.2 or later versions
+  
+- Apache jclouds: 2.2.0 or later versions
+
+### Step
+
+This example uses Pulsar 2.5.1.
+
+1. Download Pulsar tarball using one of the following ways:
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-aws.md:23]:
```suggestion
1. Download the Pulsar tarball using one of the following ways:
```

File: site2/docs/tiered-storage-aws.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+Namespace policy can be configured to offload data automatically once a threshold is reached. The threshold is based on the size of data that a topic has stored on a Pulsar cluster. Once the topic reaches the threshold, an offloading operation is triggered automatically. 
+
+Threshold value|Action
+|---|---
+> 0 | It triggers the offloading operation if the topic storage reaches its threshold.
+= 0|It causes a broker to offload data as soon as possible.
+< 0 |It disables automatic offloading operation.
+
+Automatic offloading runs when a new segment is added to a topic log. If you set the threshold on a namespace, but few messages are being produced to the topic, offload does not work until the current segment is full.
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-aws.md:191]: offload or offloader?

File: site2/docs/tiered-storage-filesystem.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+- Hadoop: 3.x.x
+
+### Step
+
+This example uses Pulsar 2.5.1.
+
+1. Download Pulsar tarball using one of the following ways:
+
+   * download the Pulsar tarball from the [Apache mirror](https://archive.apache.org/dist/pulsar/pulsar-2.5.1/apache-pulsar-2.5.1-bin.tar.gz)
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-filesystem.md:25]: same comments as that in the AWS doc.

File: site2/docs/tiered-storage-gcs.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+  
+- Apache jclouds: 2.2.0 or later versions
+
+### Step
+
+This example uses Pulsar 2.5.1.
+
+1. Download Pulsar tarball using one of the following ways:
+
+   * download the Pulsar tarball from the [Apache mirror](https://archive.apache.org/dist/pulsar/pulsar-2.5.1/apache-pulsar-2.5.1-bin.tar.gz)
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-gcs.md:25]: same comments

File: site2/docs/tiered-storage-gcs.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+> #### Tip
+>
+> For more information about the `pulsarctl namespaces set-offload-threshold options` command, including flags, descriptions, default values, and shorthands, see [here](https://streamnative.io/docs/pulsarctl/v0.4.0/#-em-set-offload-threshold-em-). 
+
+### Configure GCS offloader to run manually
+
+For individual topics, you can trigger GCS offloader manually using one of the following methods:
+
+- Use REST endpoint
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-gcs.md:206]:
```suggestion
- Use REST endpoint.
```

File: site2/docs/tiered-storage-overview.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+* Tiered storage uses [Apache Hadoop](http://hadoop.apache.org/) to support filesystems for long term storage. 
+
+    With Hadoop, it is easy to add support for more filesystems in the future.
+
+    > #### Tip
+    > 
+    > For more information about how to use the filesystem offloader with Pulsar, see [here](tiered-storage-filesystem.md).
+
+## When should I use tiered storage?
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-overview.md:29]: how about change the title to "when to use tiered storage?"

File: site2/docs/tiered-storage-overview.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+Tiered storage should be used when you have a topic for which you want to keep a very long backlog for a long time.
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-overview.md:31]: how about to change it to "Tiered storage should be used when you want to keep a very long backlog for a topic."

File: site2/docs/tiered-storage-overview.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+## When should I use tiered storage?
+
+Tiered storage should be used when you have a topic for which you want to keep a very long backlog for a long time. 
+
+For example, if you have a topic containing user actions which you use to train your recommendation systems, you may want to keep that data for a long time, so that if you change your recommendation algorithm, you can rerun it against your full user history.
+
+## How does tiered storage work?
+
+A topic in Pulsar is backed by a **log**, known as a **managed ledger**. This log is composed of an ordered list of segments. Pulsar only writes to the final segment of the log. All previous segments are sealed. The data within the segment is immutable. This is known as a **segment oriented architecture**.
```
Comment (Huanli-Meng) [inline site2/docs/tiered-storage-overview.md:37]:
```suggestion
A topic in Pulsar is backed by a **log**, known as a **managed ledger**. This log consists of an ordered list of segments. Pulsar only writes to the final segment of the log. All previous segments are sealed. The data within the segment is immutable. This is known as a **segment oriented architecture**.
```

File: site2/docs/tiered-storage-aws.md [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+
+## Installation
+
+Follow the steps below to install the AWS S3 offloader.
+
+### Prerequisite
+
+- Pulsar: 2.4.2 or later versions
+  
+- Apache jclouds: 2.2.0 or later versions
```
Comment (gaoran10) [inline site2/docs/tiered-storage-aws.md:17]: Which version of JClouds to use up to the Pulsar, so users could don't care about it.

File: site2/docs/tiered-storage-filesystem.md [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+
+## Installation
+
+Follow the steps below to install the filesystem offloader.
+
+### Prerequisite
+
+- Pulsar: 2.4.2 or later versions
+
+- Hadoop: 3.x.x
```
Comment (gaoran10) [inline site2/docs/tiered-storage-filesystem.md:17]: Same as JClouds, users could don't care about this.

File: site2/docs/tiered-storage-gcs.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+
+## Installation
+
+Follow the steps below to install the GCS offloader.
+
+### Prerequisite
+
+- Pulsar: 2.4.2 or later versions
+  
+- Apache jclouds: 2.2.0 or later versions
```
Comment (gaoran10) [inline site2/docs/tiered-storage-gcs.md:17]: Same as tiered-storage-aws.

File: site2/docs/tiered-storage-gcs.md (hunk 6 above)
Comment (Anonymitaet) [reply site2/docs/tiered-storage-gcs.md:206]: This is not a complete sentence which has a subject and predicate. 

IBM style guide:
Do not use a period after a phrase and use a period after a sentence.
image

File: site2/docs/tiered-storage-aws.md (hunk 3 above)
Comment (Anonymitaet) [reply site2/docs/tiered-storage-aws.md:191]: Offloader. Thanks for your reminding.

File: site2/docs/tiered-storage-aws.md (hunk 10 above)
Comment (Anonymitaet) [reply site2/docs/tiered-storage-aws.md:17]: Thanks. Confirmed w/ @gaoran10, users do not care about the Apache jcloud version, so I've deleted this.

File: site2/docs/tiered-storage-filesystem.md [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+
+You can configure filesystem offloader driver in the configuration file `broker.conf` or `standalone.conf`.
+
+- **Required** configurations are as below.
+  
+    Required configuration | Description | Example value
+    |---|---|---
+    `managedLedgerOffloadDriver` | Offloader driver name, which is case-insensitive. | filesystem
+    `fileSystemURI` | Connection address | hdfs://127.0.0.1:9000
+    `offloadersDirectory` | Hadoop profile path | ../conf/filesystem_offload_core_site.xml
```
Comment (gaoran10) [inline site2/docs/tiered-storage-filesystem.md:88]:
```suggestion
    `offloadersDirectory` | Offloader directory | offloaders
```

**Pull request merged**